stats/condition_interceptor.go

```diff
@@ -0,0 +1,185 @@
+// and updates the Synthetic Mass condition statistics based on the condition's patient's address.
+type ConditionStatsCreateInterceptor struct {
+	PgDataAccess    StatsDataAccess
+	MongoDataAccess server.DataAccessLayer
+}
+
+// NewConditionStatsCreateInterceptor returns an initialized instance of a ConditionStatsCreateInterceptor
+func NewConditionStatsCreateInterceptor(pgDataAccess StatsDataAccess, mongoDataAccess server.DataAccessLayer) *ConditionStatsCreateInterceptor {
+	return &ConditionStatsCreateInterceptor{PgDataAccess: pgDataAccess, MongoDataAccess: mongoDataAccess}
+}
+
+// Before is unused
+func (s *ConditionStatsCreateInterceptor) Before(resource interface{}) {}
+
+// After increments disease statistics for given after a new condition resource is created.
+func (s *ConditionStatsCreateInterceptor) After(resource interface{}) {
+	condition, ok := resource.(*models.Condition)
+
+	if ok {
+
+		var err error
+
+		if condition.Subject == nil {
+			log.Printf("ConditionStatsCreateInterceptor: Condition %s has no subject\n", condition.Id)
+			return
+		}
+
+		result, err := s.MongoDataAccess.Get(condition.Subject.ReferencedID, "Patient")
+
+		if err != nil {
+			log.Printf("ConditionStatsCreateInterceptor: Failed to get patient for condition %s\n", condition.Id)
+			return
+		}
+
+		if !conditionIsAbated(condition) {
+			patient := result.(*models.Patient)
+			err = s.PgDataAccess.AddConditionStat(patient, condition)
+		}
+
+		if err != nil {
+			log.Printf("ConditionStatsCreateInterceptor: %s\n", err.Error())
+		}
+	}
+}
+
+// OnError is unused
+func (s *ConditionStatsCreateInterceptor) OnError(err error, resource interface{}) {}
+
+// ConditionStatsUpdateInterceptor intercepts any updates to condition resources in the database
+// and updates the Synthetic Mass condition statistics based on the condition's patient's address.
+type ConditionStatsUpdateInterceptor struct {
+	PgDataAccess    StatsDataAccess
+	MongoDataAccess server.DataAccessLayer
+	// The state of the condition before the database update, for comparison after the database update
+	conditionsBefore map[string]*models.Condition
+	// Tracks if the interceptor failed to cache the condition model before the update
+	cacheError error
+}
+
+// NewConditionStatsUpdateInterceptor returns an initialized instance of ConditionStatsUpdateInterceptor
+func NewConditionStatsUpdateInterceptor(pgDataAccess StatsDataAccess, mongoDataAccess server.DataAccessLayer) *ConditionStatsUpdateInterceptor {
+	interceptor := &ConditionStatsUpdateInterceptor{PgDataAccess: pgDataAccess, MongoDataAccess: mongoDataAccess}
+	interceptor.conditionsBefore = make(map[string]*models.Condition)
+	return interceptor
+}
+
+// Before caches a condition resource before it's updated, for comparison after the update.
+func (s *ConditionStatsUpdateInterceptor) Before(resource interface{}) {
+	condition, ok := resource.(*models.Condition)
+
+	if ok {
+		s.conditionsBefore[condition.Id] = condition
```

Comment:
As you noted in your reply to my previous comments, you probably should be synchronizing on your writes and deletes to the map, right?  Same for the map in the patient interceptor.